core/extension/gdextension_interface.h

```diff
@@ -283,6 +285,12 @@ typedef struct {
 	GDExtensionClassCreateInstance create_instance_func; // (Default) constructor; mandatory. If the class is not instantiable, consider making it virtual or abstract.
 	GDExtensionClassFreeInstance free_instance_func; // Destructor; mandatory.
 	GDExtensionClassGetVirtual get_virtual_func; // Queries a virtual function by name and returns a callback to invoke the requested virtual function.
+	// Can be used instead of `get_virtual_func`. Returns user data (non-null) that will be passed back when calling `call_virtual_func`.
+	// Returning `null` from this function signals to Godot that the virtual function is not overridden.
+	// Data provided to this function should be managed by the extension and must live until the GDExtensionInstance.
```

Comment:
What about:

```suggestion
	// Can be used instead of `get_virtual_func`. Returns user data that will be passed to `call_virtual_func`.
	// Returning `NULL` from this function signals to Godot that the virtual function is not overridden.
	// Data returned from this function should be managed by the extension and must be valid until the extension is deinitialized.
```

I got a little carried away and made a whole bunch of changes that I think simplify the language a bit :-)

But the truly important change is end of the last sentence: "and must be valid until the extension is deinitialized"